src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableSchemaObject.java

```diff
@@ -115,4 +127,127 @@ public static TableSchemaObject from(TableMetadata tableMetadata, ObjectMapper o
     }
     return new TableSchemaObject(tableMetadata, vectorConfig);
   }
+
+  /**
+   * Convert table schema object to table response which is returned as response for `listTables`
+   *
+   * @return
+   */
+  public TableResponse toTableResponse() {
+    String tableName = removeQuotes(tableMetadata().getName());
+    HashMap<String, ColumnType> columnsDefinition = new HashMap<>();
+    for (Map.Entry<CqlIdentifier, ColumnMetadata> column :
+        tableMetadata().getColumns().entrySet()) {
+      ColumnType type = getColumnType(column.getKey().asInternal(), column.getValue());
+      columnsDefinition.put(removeQuotes(column.getKey()), type);
+    }
+
+    final List<String> partitionBy =
+        tableMetadata().getPartitionKey().stream()
+            .map(column -> removeQuotes(column.getName()))
+            .collect(Collectors.toList());
+    final List<PrimaryKey.OrderingKey> partitionSort =
+        tableMetadata().getClusteringColumns().entrySet().stream()
+            .map(
+                entry ->
+                    new PrimaryKey.OrderingKey(
+                        removeQuotes(entry.getKey().getName()),
+                        entry.getValue() == ClusteringOrder.ASC
+                            ? PrimaryKey.OrderingKey.Order.ASC
+                            : PrimaryKey.OrderingKey.Order.DESC))
+            .collect(Collectors.toList());
+    PrimaryKey primaryKey =
+        new PrimaryKey(
+            partitionBy.toArray(new String[0]),
+            partitionSort.toArray(new PrimaryKey.OrderingKey[0]));
+    return new TableResponse(
+        tableName, new TableResponse.TableDefinition(columnsDefinition, primaryKey));
+  }
+
+  // Need to handle frozen types
```

Comment:
Should maybe expand here... handle how? (I know we only support use of non-frozen types)

Reply:
Handled them, comment from Friday, missed to remove.